File: LeetCode/28.find-the-index-of-the-first-occurrence-in-a-string.cpp
/*
 * @lc app=leetcode id=28 lang=cpp
 *
 * [28] Find the Index of the First Occurrence in a String
 */
#include <iostream>

#include <string>

using namespace std;

// @lc code=start
class Solution {
public:

int strStr(string haystack, string needle)
{
    if (needle.size() > haystack.size())
    {
        return -1;
    }
    int i = 0;
    while (i < haystack.size() &&(haystack.size() -  (i > needle.size())))
    {
        if ((haystack.size() - i) < needle.size())
        {
            return -1;
        }
        string f = haystack.substr(i,  needle.size());
        if (f == needle)
        {
            return i;
        }
        i++;
    }

    return -1;
}
};
// @lc code=end

File: LeetCode/28.find-the-index-of-the-first-occurrence-in-a-string.py
#
# @lc app=leetcode id=28 lang=python3
#
# [28] Find the Index of the First Occurrence in a String
#

# @lc code=start
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        
        for idx in range(len(haystack) - len(needle)  +1):
            sub_string = haystack[idx:idx+len(needle)]
            if needle == sub_string:return idx
        return -1

# @lc code=end

sol = Solution()
haystack = "hello"
needle = "ll"

print(sol.strStr(haystack, needle))